starknet_py/abi/v1/parser_transformer.py

```diff
@@ -0,0 +1,120 @@
+import os
+from typing import Any, List, Optional
+
+import lark
+from lark import Token, Transformer, v_args
+
+from starknet_py.cairo.data_types import (
+    ArrayType,
+    CairoType,
+    FeltType,
+    OptionType,
+    TupleType,
+    TypeIdentifier,
+    UintType,
+)
+
+
+class ParserTransformer(Transformer):
+    """
+    Transforms the lark tree into CairoTypes.
+    """
+
+    # pylint: disable=no-self-use
+
+    def __default__(self, data: str, children, meta):
+        raise TypeError(f"Unable to parse tree node of type {data}")
+
+    @v_args(inline=True)
+    def start(self, value) -> Optional[CairoType]:
+        """
+        Method all the entries starts from.
+        """
+        return value
+
+    def type(self, value: List[Optional[CairoType]]) -> Optional[CairoType]:
+        """
+        Types starting with "core" or unit type "()".
+
+        Tokens are read bottom-up, so here all of them are parsed and should be just returned.
+        `Optional` is added in case of the unit type.
+        """
+        assert len(value) == 1
+        return value[0]
+
+    def type_felt(self, _value: List[Any]) -> FeltType:
+        """
+        Felt does not contain any additional arguments, so `_value` is just an empty list.
+        """
+        return FeltType()
+
+    def type_uint(self, value: List[Token]) -> UintType:
+        """
+        Uint type contains information about it's size. It is present in the value[0].
+        """
+        return UintType(int(value[0]))
+
+    def type_unit(self, _value: List[Any]) -> None:
+        """
+        () type.
```

Comment:
So just empty type or?

Reply:
I think the unit type seems to be `void` in the cairo1.0